File: BE/src/main/java/com/haivn/common_api/XeLuuBai.java
package com.haivn.common_api;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.Where;

import javax.persistence.*;

@Entity
@Table(name = "xe_luu_bai")
@Getter
@Setter
@DynamicUpdate
@Where(clause = "deleted=false")
public class XeLuuBai extends BaseEntity{
    @Column(name = "idNvT")
    private Long idNvT;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="idNvT",referencedColumnName="id", nullable = false, insertable = false, updatable = false)
    private NguoiDung nhanVienThem;
    @Column(name = "xe")
    private Short xe;
    @Column(name = "bien_so")
    private String bienSo;
    @Column(name = "mo_ta")
    private String moTa;
    @Column(name = "idNvXN")
    private Long idNvXN;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="idNvXN",referencedColumnName="id", nullable = false, insertable = false, updatable = false)
    private NguoiDung nhanVienXacNhan;
    @Column(name = "ly_do")
    private String lyDo;
    @Column(name = "idBx")
    private Long idBx;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="idBx",referencedColumnName="id", nullable = false, insertable = false, updatable = false)
    private BaiXe baiXe;
    @Column(name = "status")
    private Short status;
}

File: fe/lib/models/xe-luu-bai.dart
import 'bai-xe.dart';
import 'user.dart';

class XeLuuBai {
  String? createdDate;
  String? modifiedDate;
  int? id;
  int? idNvT;
  User? nhanVienThem;
  int? xe;
  String? bienSo;
  String? moTa;
  int? idNvXN;
  User? nhanVienXacNhan;
  String? lyDo;
  int? idBx;
  BaiXe? baiXe;
  int? status;
  XeLuuBai({
    this.id,
    this.createdDate,
    this.idBx,
    this.baiXe,
    this.modifiedDate,
    this.idNvT,
    this.nhanVienThem,
    this.xe,
    this.bienSo,
    this.moTa,
    this.idNvXN,
    this.nhanVienXacNhan,
    this.lyDo,
    this.status,
  });
  XeLuuBai.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        idNvT = json['idNvT'],
        createdDate = json['createdDate'],
        modifiedDate = json['modifiedDate'],
        nhanVienThem = User.fromJson(json['nhanVienThem']),
        idNvXN = json['idNvT'],
        nhanVienXacNhan = (json['nhanVienXacNhan'] != null)
            ? User.fromJson(json['nhanVienXacNhan'])
            : null,
        xe = json['xe'],
        bienSo = json['bienSo'],
        lyDo = json['lyDo'],
        moTa = json['moTa'],
        idBx = json['idBx'],
        baiXe = (json['baiXe'] != null) ? BaiXe.fromJson(json['baiXe']) : null,
        status = json['status'];
  Map<String, dynamic> toJson() => {
        'idNvT': idNvT,
        'xe': xe,
        'bienSo': bienSo,
        'moTa': moTa,
        'idBx': idBx,
        'idNvXN': idNvXN,
        'lyDo': lyDo,
        'status': status,
      };
}
